Addition of magimage function

AngusWright opened the pull request:
This function is a replacement for the graphics ‘image’ function, and
incorporates magmap scaling and magaxis axes.

Commit:
Addition of magimage function

This function is a replacement for the graphics ‘image’ function, and
incorporates magmap scaling and magaxis axes.

asgr commented:
Thanks, I shall take a look!

Cheers,

Aaron

On 10 Jan 2015, at 4:10 pm, AngusWright <notifications@github.com<mailto:notifications@github.com>> wrote:

This function is a replacement for the graphics ‘image’ function, and
incorporates magmap scaling and magaxis axes.

---

You can merge this Pull Request by running

  git pull https://github.com/AngusWright/magicaxis master

Or view, comment on, or merge it at:

  https://github.com/asgr/magicaxis/pull/1

Commit Summary
-   Addition of magimage function

File Changes
-   A R/magimage.Rhttps://github.com/asgr/magicaxis/pull/1/files#diff-0 (42)
-   A man/magimage.Rdhttps://github.com/asgr/magicaxis/pull/1/files#diff-1 (60)

Patch Links:
-   https://github.com/asgr/magicaxis/pull/1.patch
-   https://github.com/asgr/magicaxis/pull/1.diff

—
Reply to this email directly or view it on GitHubhttps://github.com/asgr/magicaxis/pull/1.

Commit:
Removed hard-coding of labels and side args

Removed hard-coding of labels and side args and corrected the
documentation to reflect the change

AngusWright commented:
Update after most recent commit; removed the hard coding of some options. Modified documentation to reflect the change.

asgr commented:
I don't think I can merge this quite yet, although it looks like a useful addition (so I'm keen to). My basic philosophy with magX functions is the "..." should pass to the X like function that we're replacing. i.e. in magplot these go to plot and in magcon they go to contour. In this case it should pass to image. This probably means you need to specify more of the magmap options explicitly. Do you see my point?

AngusWright commented:
Absolutely. I just figured that since the options to image are fairly useless that magmap options were more suited, but no matter. Happy to conform to the standard. I'll update and re-commit

asgr commented:
Also, I think "side" should be 1:2 by default (consistent with magplot and base image).

asgr commented on R/magimage.R:
```diff
@@ -0,0 +1,42 @@
+magimage<-function(mat, astr=NULL, main="", xlab=ifelse(is.null(astr),"X (pix)","RA (deg)"), ylab=ifelse(is.null(astr),"Y (pix)","Dec (deg)"), side=1:4,labels=c(T,T,F,F),...) {
```
I'd move "astr" much further down the paramter list I think- it's quite unfamiliar to people who are using this instead of base image.

AngusWright replied:
Agreed

asgr commented:
Hmm, you might have a point. What image options are left to be used?

AngusWright commented:
There are actually quite a few now that I look, I've just never really used them! 

x=          y=          zlim=       ylim=       add=        yaxs=       ylab=       oldstyle=   cex=  
...=        z=          xlim=       col=        xaxs=       xlab=       breaks=     useRaster=

asgr commented:
Basically, unless i"m missing something key I'd copy what I did with magplot (it has all the magaxis options and sends the ... to plot). This wasn't always the case, but I got a feature request from somebody a while back, and I figured they were right and I was lazy :-P

AngusWright commented:
Fair enough. I'll make these changes tomorrow, just getting some other stuff done and making the original change was part of the process, so I figured I'd gitpush for posterity. :+1:

asgr commented:
Superseded with my own magimage function.
